# Insights
## Text Analysis of "THE ALCHEMIST" by Paulo Coelho
- In this Jupyter Notebook i am going to do text analysis by using NLP
### Index:
- Reading text file.
- Conversion of text in TextBlob type.
- lowering all text.
- counting repeatation of words.
- N-gram (3 consecutive words)
- POS (Part of Speech tagging)
- Polarity of sentence
- - Printing each sentence of entire book with polarity score (-ve sentiment or +ve sentiment)
- - Printing only Positive polarity sentences with polarity score.
- - Printing only Negative polarity sentences with polarity score.


# Importing Textblob

In [72]:
from textblob import TextBlob

### Reading Text file

In [73]:
text = open('THE_ALCHEMIST.txt',encoding="utf8")

In [74]:
text = text.read()

### Conversion of text in TextBlob type.

In [33]:
blob = TextBlob(text)

In [122]:
len(blob)

81643

# Out of 81643 words considering only 5000 words to avoid more scrolling

In [128]:
blob = blob[0:5000]

In [129]:
print(len(blob))

5000


### lowering all text.

In [130]:
blob = blob.lower()

In [132]:
print(blob)

there are some books that go beyond being special. jonathan livingston seagull, the
little prince, conversations with god and the prophet would all make the list. i’d like
to add one more, the alchemist by paulo coelho. it is the story of santiago, a shepard in
spain. he sets off to find a treasure by the pyramids in egypt that is spoken of in a
reoccurring dream. a story as old as the pyramids themselves. along the way, he finds
out that his journey is about discovering his personal legend, touching the soul of the
world, and learning what love is really about; as well as getting the loot. we learn as we
travel with santiago many of the valuable lessons of life, and uncover many of the
mysteries. “the wise men understood that this natural world is only an image and a copy
of paradise. the existence of this world is simply a guarantee that there exists a world
that is perfect. god created the world so that, through its visible objects, men could
understand his spiritual teachings and t

### Separating all text in to sentences

In [136]:
blob_sentences = blob.sentences

In [137]:
blob_sentences

[Sentence("there are some books that go beyond being special."),
 Sentence("jonathan livingston seagull, the
 little prince, conversations with god and the prophet would all make the list."),
 Sentence("i’d like
 to add one more, the alchemist by paulo coelho."),
 Sentence("it is the story of santiago, a shepard in
 spain."),
 Sentence("he sets off to find a treasure by the pyramids in egypt that is spoken of in a
 reoccurring dream."),
 Sentence("a story as old as the pyramids themselves."),
 Sentence("along the way, he finds
 out that his journey is about discovering his personal legend, touching the soul of the
 world, and learning what love is really about; as well as getting the loot."),
 Sentence("we learn as we
 travel with santiago many of the valuable lessons of life, and uncover many of the
 mysteries."),
 Sentence("“the wise men understood that this natural world is only an image and a copy
 of paradise."),
 Sentence("the existence of this world is simply a guarantee that th

### Total 44 sentence with 5000 words

In [139]:
len(blob_sentences)

44

### counting repeatation of words.

In [163]:
blob.word_counts

defaultdict(int,
            {'there': 3,
             'are': 3,
             'some': 3,
             'books': 2,
             'that': 23,
             'go': 1,
             'beyond': 1,
             'being': 1,
             'special': 1,
             'jonathan': 1,
             'livingston': 1,
             'seagull': 1,
             'the': 82,
             'little': 1,
             'prince': 1,
             'conversations': 1,
             'with': 4,
             'god': 2,
             'and': 22,
             'prophet': 1,
             'would': 1,
             'all': 6,
             'make': 1,
             'list': 1,
             'i': 9,
             '’': 9,
             'd': 1,
             'like': 2,
             'to': 28,
             'add': 1,
             'one': 3,
             'more': 2,
             'alchemist': 4,
             'by': 4,
             'paulo': 1,
             'coelho': 1,
             'it': 9,
             'is': 8,
             'story': 5,
             'of': 25,

### N-gram (3 consecutive words)

In [150]:
# Printing only 100 worslist to avoid more scrolling

blob.ngrams(3)[0:100]

[WordList(['there', 'are', 'some']),
 WordList(['are', 'some', 'books']),
 WordList(['some', 'books', 'that']),
 WordList(['books', 'that', 'go']),
 WordList(['that', 'go', 'beyond']),
 WordList(['go', 'beyond', 'being']),
 WordList(['beyond', 'being', 'special']),
 WordList(['being', 'special', 'jonathan']),
 WordList(['special', 'jonathan', 'livingston']),
 WordList(['jonathan', 'livingston', 'seagull']),
 WordList(['livingston', 'seagull', 'the']),
 WordList(['seagull', 'the', 'little']),
 WordList(['the', 'little', 'prince']),
 WordList(['little', 'prince', 'conversations']),
 WordList(['prince', 'conversations', 'with']),
 WordList(['conversations', 'with', 'god']),
 WordList(['with', 'god', 'and']),
 WordList(['god', 'and', 'the']),
 WordList(['and', 'the', 'prophet']),
 WordList(['the', 'prophet', 'would']),
 WordList(['prophet', 'would', 'all']),
 WordList(['would', 'all', 'make']),
 WordList(['all', 'make', 'the']),
 WordList(['make', 'the', 'list']),
 WordList(['the', 'list',

## POS (Part of Speech tagging)

In [153]:
# Printing only 100 worslist to avoid more scrolling

blob.pos_tags[0:100]

[('there', 'EX'),
 ('are', 'VBP'),
 ('some', 'DT'),
 ('books', 'NNS'),
 ('that', 'IN'),
 ('go', 'VBP'),
 ('beyond', 'IN'),
 ('being', 'VBG'),
 ('special', 'JJ'),
 ('jonathan', 'NN'),
 ('livingston', 'NN'),
 ('seagull', 'NN'),
 ('the', 'DT'),
 ('little', 'JJ'),
 ('prince', 'NN'),
 ('conversations', 'NNS'),
 ('with', 'IN'),
 ('god', 'NN'),
 ('and', 'CC'),
 ('the', 'DT'),
 ('prophet', 'NN'),
 ('would', 'MD'),
 ('all', 'DT'),
 ('make', 'VB'),
 ('the', 'DT'),
 ('list', 'NN'),
 ('i', 'NN'),
 ('’', 'VBP'),
 ('d', 'NN'),
 ('like', 'IN'),
 ('to', 'TO'),
 ('add', 'VB'),
 ('one', 'CD'),
 ('more', 'JJR'),
 ('the', 'DT'),
 ('alchemist', 'NN'),
 ('by', 'IN'),
 ('paulo', 'NN'),
 ('coelho', 'NN'),
 ('it', 'PRP'),
 ('is', 'VBZ'),
 ('the', 'DT'),
 ('story', 'NN'),
 ('of', 'IN'),
 ('santiago', 'NN'),
 ('a', 'DT'),
 ('shepard', 'NN'),
 ('in', 'IN'),
 ('spain', 'NN'),
 ('he', 'PRP'),
 ('sets', 'VBZ'),
 ('off', 'IN'),
 ('to', 'TO'),
 ('find', 'VB'),
 ('a', 'DT'),
 ('treasure', 'NN'),
 ('by', 'IN'),
 ('the',

## Printing each sentence of entire book with polarity score (-ve sentiment or +ve sentiment)

In [158]:
for s in blob_sentences:
    
    if s.polarity>0:
        print("----------------- Positive sentence ------------------------------------\n")
        print(f"Sentence = {s} => [Polarity = {s.polarity}] \n")
    elif s.polarity<0:
        print("----------------- Negative sentence ------------------------------------\n")
        print(f"Sentence = {s} => [Polarity = {s.polarity}] \n")

----------------- Positive sentence ------------------------------------

Sentence = there are some books that go beyond being special. => [Polarity = 0.35714285714285715] 

----------------- Negative sentence ------------------------------------

Sentence = jonathan livingston seagull, the
little prince, conversations with god and the prophet would all make the list. => [Polarity = -0.1875] 

----------------- Positive sentence ------------------------------------

Sentence = i’d like
to add one more, the alchemist by paulo coelho. => [Polarity = 0.5] 

----------------- Positive sentence ------------------------------------

Sentence = a story as old as the pyramids themselves. => [Polarity = 0.1] 

----------------- Positive sentence ------------------------------------

Sentence = along the way, he finds
out that his journey is about discovering his personal legend, touching the soul of the
world, and learning what love is really about; as well as getting the loot. => [Polarity = 0

### Printing only Positive polarity sentences with polarity score.

In [159]:
print("----- +ve POLARITY SENTENCES------")
for s in blob_sentences:
     if s.polarity>0:
        print("----------------- Positive sentence ------------------------------------\n")
        print(f"Sentence = {s} => [Polarity = {s.polarity}] \n")
   

----- +ve POLARITY SENTENCES------
----------------- Positive sentence ------------------------------------

Sentence = there are some books that go beyond being special. => [Polarity = 0.35714285714285715] 

----------------- Positive sentence ------------------------------------

Sentence = i’d like
to add one more, the alchemist by paulo coelho. => [Polarity = 0.5] 

----------------- Positive sentence ------------------------------------

Sentence = a story as old as the pyramids themselves. => [Polarity = 0.1] 

----------------- Positive sentence ------------------------------------

Sentence = along the way, he finds
out that his journey is about discovering his personal legend, touching the soul of the
world, and learning what love is really about; as well as getting the loot. => [Polarity = 0.3] 

----------------- Positive sentence ------------------------------------

Sentence = we learn as we
travel with santiago many of the valuable lessons of life, and uncover many of the

### Printing only Negative polarity sentences with polarity score.

In [160]:
print("----- -ve POLARITY SENTENCES------")
for s in blob_sentences:
    
    if s.polarity<0:
        print("----------------- Negative sentence ------------------------------------\n")
        print(f"Sentence = {s} => [Polarity = {s.polarity}] \n")

----- -ve POLARITY SENTENCES------
----------------- Negative sentence ------------------------------------

Sentence = jonathan livingston seagull, the
little prince, conversations with god and the prophet would all make the list. => [Polarity = -0.1875] 

----------------- Negative sentence ------------------------------------

Sentence = he was so fascinated by himself that, one morning, he fell
into the lake and drowned. => [Polarity = -0.1] 

----------------- Negative sentence ------------------------------------

Sentence = the roof had fallen in long ago, and an enormous sycamore had
grown on the spot where the sacristy had once stood. => [Polarity = -0.025] 

----------------- Negative sentence ------------------------------------

Sentence = during the
two hours that they talked, she told him she was the merchant’s daughter, and spoke of
life in the village, where each day was like all the others, the shepherd told her of the
andalusian countryside, and related the news from 